<!--
  Copyright (C) 2018 LEIDOS.

  Licensed under the Apache License, Version 2.0 (the "License"); you may not
  use this file except in compliance with the License. You may obtain a copy of
  the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  License for the specific language governing permissions and limitations under
  the License.
-->
<!-- Launch file for the pinpoint driver -->
<launch>
  <arg name="pinpoint_address" default="192.168.88.29" doc="Pinpoint device ip address"/>
  <arg name="loc_port" default="9501" doc="The localization port the pinpoint is sending data to"/>
  <arg name="odom_frame" default="odom" doc="The name of the local planning frame"/>
  <arg name="base_link_frame" default="base_link" doc="The name of the robot base frame"/>
  <arg name="sensor_frame" default="pinpoint" doc="The name of the pinpoint device frame"/>

  <node pkg="pinpoint" type="pinpoint_node" name="pin_point_application">
    <param name="address" value="$(arg address)"/>
    <param name="loc_port" value="$(arg loc_port)"/>
    <param name="odom_frame" value="$(arg odom_frame)"/>
    <param name="base_link_frame" value="$(arg base_link_frame)"/>
    <param name="sensor_frame" value="$(arg sensor_frame)"/>
  </node>
</launch>
